app/pages/project/instances/instance/tabs/MetricsTab.tsx

```diff
@@ -5,230 +5,86 @@
 
-  const { startTime, endTime, dateTimeRangePicker } = useDateTimeRangePicker({
-    initialPreset: 'lastDay',
-  })
-
-  // The fallback here is kind of silly — it is only invoked when there are no
-  // disks, in which case we show the fallback UI and diskName is never used. We
-  // only need to do it this way because hooks cannot be called conditionally.
-  const [diskName, setDiskName] = useState<string>(disks[0]?.name || '')
-  const diskItems = disks.map(({ name }) => ({ label: name, value: name }))
-
-  if (disks.length === 0) {
-    return (
-      <TableEmptyBox>
-        <EmptyMessage
-          icon={<Storage24Icon />}
-          title="No metrics available"
-          body="Metrics are only available if there are disks attached"
-        />
-      </TableEmptyBox>
-    )
-  }
-
-  const commonProps = {
+  // memoizing here would be redundant because the only things that cause a
+  // render are these values, which would all be in the dep array anyway
+  const context = {
     startTime,
     endTime,
-    diskSelector: { project, disk: diskName },
+    dateTimeRangePicker,
+    intervalPicker,
+    setIsIntervalPickerEnabled,
   }
 
+  // Find the relevant <Outlet> in RouteTabs
   return (
-    <>
-      <div className="mb-4 flex justify-between">
-        <Listbox
-          className="w-64"
-          aria-label="Choose disk"
-          name="disk-name"
-          selected={diskName}
-          items={diskItems}
-          onChange={(val) => {
-            if (val) setDiskName(val)
-          }}
-        />
-        {dateTimeRangePicker}
-      </div>
-
-      <div className="mt-8 space-y-12">
-        {/* see the following link for the source of truth on what these mean
-            https://github.com/oxidecomputer/crucible/blob/258f162b/upstairs/src/stats.rs#L9-L50 */}
-        <div className="flex w-full space-x-4">
-          <DiskMetric {...commonProps} title="Reads" unit="Count" metric="read" />
-          <DiskMetric {...commonProps} title="Read" unit="Bytes" metric="read_bytes" />
-        </div>
-
-        <div className="flex w-full space-x-4">
-          <DiskMetric {...commonProps} title="Writes" unit="Count" metric="write" />
-          <DiskMetric {...commonProps} title="Write" unit="Bytes" metric="write_bytes" />
-        </div>
-
-        <div className="flex w-full space-x-4">
-          <DiskMetric {...commonProps} title="Flushes" unit="Count" metric="flush" />
-        </div>
-      </div>
-    </>
+    <MetricsContext.Provider value={context}>
+      <RouteTabs sideTabs tabListClassName="mt-24">
+        <Tab to={pb.instanceCpuMetrics({ project, instance })} sideTab>
+          CPU
+        </Tab>
+        <Tab to={pb.instanceDiskMetrics({ project, instance })} sideTab>
+          Disk
+        </Tab>
+        <Tab to={pb.instanceNetworkMetrics({ project, instance })} sideTab>
+          Network
+        </Tab>
+      </RouteTabs>
+    </MetricsContext.Provider>
```

Comment:
Feels a bit silly to have to put `sideTabs` on both the parent and the children. We could fix this either by taking the list of entries as a prop on `<RouteTabs>` and having it render the children and passing through the right prop, OR we could do it entirely in CSS, allowing the class on the parent to determine the styling on the children. I think either of those would be fine, but because they require changing all existing `RouteTabs` call sites, let's make it a followup.